framework/test/jest/specs/unit/controller/channels/child_process_channel.spec.js

```diff
@@ -0,0 +1,622 @@
+const axon = require('axon');
+const axonRpc = require('axon-rpc');
+
+jest.mock('eventemitter2');
+jest.mock('../../../../../../src/controller/event', () =>
+	jest.fn().mockImplementation(name => ({
+		name,
+		module: 'moduleAlias',
+		key: () => 'aKey',
+		serialize: () => 'serialized',
+	}))
+);
+
+axonRpc.Client = function() {
+	return {
+		call: jest.fn(),
+	};
+};
+axonRpc.Server = function() {
+	return {
+		expose: jest.fn(),
+	};
+};
+
+const EventEmitter2 = require('eventemitter2');
+const Event = require('../../../../../../src/controller/event');
+const ChildProcessChannel = require('../../../../../../src/controller/channels/child_process_channel');
+const BaseChannel = require('../../../../../../src/controller/channels/base_channel');
+
+describe('ChildProcessChannel Channel', () => {
+	const originalModuleAlias = 'moduleAlias';
+	const params = {
+		moduleAlias: originalModuleAlias,
+		events: ['event1', 'event2'],
+		actions: {
+			action1: jest.fn(),
+			action2: jest.fn(),
+			action3: jest.fn(),
+		},
+		options: {},
+	};
+	let childProcessChannel;
+	const socketsPath = {
+		root: 'root',
+		sub: 'sub',
+		pub: 'pub',
+		rpc: 'rpc',
+	};
+
+	let resolveWhenAllSocketsBound;
+	let rejectWhenAnySocketFailsToBind;
+	let rejectWhenTimeout;
+	let removeAllListeners;
+
+	beforeEach(() => {
```

Comment:
should we consistently use `async` for `it`, 'beforeEach` and `afterEach`?

Reply:
I think there was an `eslint` rule for Mocha tests but not for Jest, that was forcing all the tests to be asynchronous.  I am not really sure if the same needs to be applied to `Jest`. Maybe @yatki can throw some light here. In my point of view, if a test doesn't contain asynchronous code, defining the function as async makes it misleading. This scenario is different from when we write `async` in a function definition so the user is aware that this function returns a promise.

Reply:
yea, I think that was my first thought too.
https://github.com/lo1tuma/eslint-plugin-mocha/blob/master/docs/rules/no-synchronous-tests.md
but it's more like safety measure for unstable test

Reply:
@shuse2 @limiaspasdaniel It's definitely not needed to use `async` keyword. 

- You need `async` when you need to await a promise.
- You need `done` when you need assertions inside a callback.

There is one important info that we all need to be careful that is using `resolves` / `rejects` keyword: https://jestjs.io/docs/en/expect#resolves